Build failure: new-lg4ff with kernel 6.15 and now also  6.14

### Nixpkgs version

- Unstable (25.11)

### Steps to reproduce

1. Use a Kernel with version 6.15 (or some version of 6.14)
2. enable the "hardware.new-lg4ff.enable" option
3. see the build failing

### Can Hydra reproduce this build failure?

Yes, Hydra can reproduce this build failure.

### Link to Hydra build job

https://hydra.nixos.org/build/298317206

### Relevant log output

```console
error: builder for '/nix/store/fmmkgs75kcjfinahgx3nr9xqw5v90dr6-new-lg4ff-0-unstable-2024-11-25.drv' failed with exit code 2;
       last 25 log lines:
       > make[1]: Entering directory '/nix/store/6lcz8r0ygva68y2iy6b1idqypc5lv0hq-linux-6.15.0-dev/lib/modules/6.15.0-cachyos/build'
       > make[2]: Entering directory '/build/source'
       >   CC [M]  hid-lg.o
       >   CC [M]  hid-lg4ff.o
       > hid-lg4ff.c:462:6: warning: no previous prototype for 'lg4ff_send_cmd_with_id' []
       >   462 | void lg4ff_send_cmd_with_id(struct lg4ff_device_entry *entry, u8 *cmd, u8 id) {
       >       |      ^~~~~~~~~~~~~~~~~~~~~~
       > hid-lg4ff.c:480:6: warning: no previous prototype for 'lg4ff_send_cmd' []
       >   480 | void lg4ff_send_cmd(struct lg4ff_device_entry *entry, u8 *cmd)
       >       |      ^~~~~~~~~~~~~~
       > hid-lg4ff.c:498:6: warning: no previous prototype for 'lg4ff_update_slot' []
       >   498 | void lg4ff_update_slot(struct lg4ff_slot *slot, struct lg4ff_effect_parameters *parameters)
       >       |      ^~~~~~~~~~~~~~~~~
       > hid-lg4ff.c: In function 'lg4ff_init':
       > hid-lg4ff.c:2429:9: error: implicit declaration of function 'hrtimer_init'; did you mean 'hrtimers_init'? []
       >  2429 |         hrtimer_init(&entry->hrtimer, CLOCK_MONOTONIC, HRTIMER_MODE_REL);
       >       |         ^~~~~~~~~~~~
       >       |         hrtimers_init
       > make[4]: *** [/nix/store/6lcz8r0ygva68y2iy6b1idqypc5lv0hq-linux-6.15.0-dev/lib/modules/6.15.0-cachyos/source/scripts/Makefile.build:208: hid-lg4ff.o] Error 1
       > make[3]: *** [/nix/store/6lcz8r0ygva68y2iy6b1idqypc5lv0hq-linux-6.15.0-dev/lib/modules/6.15.0-cachyos/source/Makefile:2039: .] Error 2
       > make[2]: *** [/nix/store/6lcz8r0ygva68y2iy6b1idqypc5lv0hq-linux-6.15.0-dev/lib/modules/6.15.0-cachyos/source/Makefile:248: __sub-make] Error 2
       > make[2]: Leaving directory '/build/source'
       > make[1]: *** [/nix/store/6lcz8r0ygva68y2iy6b1idqypc5lv0hq-linux-6.15.0-dev/lib/modules/6.15.0-cachyos/source/Makefile:248: __sub-make] Error 2
       > make[1]: Leaving directory '/nix/store/6lcz8r0ygva68y2iy6b1idqypc5lv0hq-linux-6.15.0-dev/lib/modules/6.15.0-cachyos/build'
       > make: *** [Makefile:5: default] Error 2
```

### Additional context

That is the URL of a example hydra jobs log: [https://hydra.nixos.org/build/298317206/log](https://hydra.nixos.org/build/298317206/log)

I am not sure if it can be fixed without updating to a newer (git-) version of new-lg4ff (but there isn't a new release)
There is a recent commit in the projects git-repository which seems to fix compatibility with kernel 6.15.
[https://github.com/berarma/new-lg4ff/commit/d6c8a82892b2662d017cc301aa94e255d72e9284](https://github.com/berarma/new-lg4ff/commit/d6c8a82892b2662d017cc301aa94e255d72e9284)

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.15.0-cachyos, NixOS, 25.11 (Xantusia), 25.11.20250531.910796c`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.28.3`
 - channels(root): `"nixos"`
 - nixpkgs: `/nix/store/2q205l7a46885nq31vvji40m30bygzzj-source`

### Notify maintainers

@MatthiasBenaets 
I hope there is something you can do about that. Thank you for maintaining that package :)

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/os-specific/linux/new-lg4ff/default.nix
+++ b/pkgs/os-specific/linux/new-lg4ff/default.nix
@@ -25,19 +25,20 @@ stdenv.mkDerivation rec {
 
   nativeBuildInputs = kernel.moduleBuildDependencies;
 
-  makeFlags = [
-    "KVERSION=${kernel.modDirVersion}"
-    "KDIR=${kernel.dev}/lib/modules/${kernel.modDirVersion}/build"
-    # Fake the CONFIG_LOGIWHEELS_FF config option, so it builds for 6.15
-    "KCFLAGS=-DCONFIG_LOGIWHEELS_FF"
-  ];
+  preConfigure = ''
+    makeFlagsArray+=(
+      KVERSION="${kernel.modDirVersion}"
+      KDIR="${kernel.dev}/lib/modules/${kernel.modDirVersion}/build"
+      KCFLAGS="-DCONFIG_LOGIWHEELS_FF -DCONFIG_LEDS_CLASS"
+    )
+  '';
 
-  meta = with lib; {
+  meta = {
     description = "Experimental Logitech force feedback module for Linux";
     homepage = "https://github.com/berarma/new-lg4ff";
-    license = licenses.gpl2Only;
-    maintainers = with maintainers; [ matthiasbenaets ];
-    platforms = platforms.linux;
+    license = lib.licenses.gpl2Only;
+    maintainers = with lib.maintainers; [ matthiasbenaets ];
+    platforms = lib.platforms.linux;
     broken = stdenv.hostPlatform.isAarch64;
   };
 }

--- a/pkgs/os-specific/linux/new-lg4ff/default.nix
+++ b/pkgs/os-specific/linux/new-lg4ff/default.nix
@@ -37,7 +37,10 @@ stdenv.mkDerivation rec {
     description = "Experimental Logitech force feedback module for Linux";
     homepage = "https://github.com/berarma/new-lg4ff";
     license = lib.licenses.gpl2Only;
-    maintainers = with lib.maintainers; [ matthiasbenaets ];
+    maintainers = with lib.maintainers; [
+      amadejkastelic
+      matthiasbenaets
+    ];
     platforms = lib.platforms.linux;
     broken = stdenv.hostPlatform.isAarch64;
   };

--- a/pkgs/os-specific/linux/new-lg4ff/default.nix
+++ b/pkgs/os-specific/linux/new-lg4ff/default.nix
@@ -5,15 +5,15 @@
   fetchFromGitHub,
 }:
 
-stdenv.mkDerivation {
+stdenv.mkDerivation rec {
   pname = "new-lg4ff";
-  version = "0-unstable-2024-11-25";
+  version = "0.5.0";
 
   src = fetchFromGitHub {
     owner = "berarma";
     repo = "new-lg4ff";
-    rev = "6100a34c182536c607af80e119d54a66c6fb2a23";
-    sha256 = "sha256-90PnQDGwp94ELvWx6p8QiZucYmTbH3N0GiZbj3fo25g=";
+    tag = "v${version}";
+    sha256 = "sha256-nh5J89S3z0odzh2fDsAVVY1X6lr4ZUwoyu3UVOYQiq8=";
   };
 
   preBuild = ''
@@ -28,6 +28,8 @@ stdenv.mkDerivation {
   makeFlags = [
     "KVERSION=${kernel.modDirVersion}"
     "KDIR=${kernel.dev}/lib/modules/${kernel.modDirVersion}/build"
+    # Fake the CONFIG_LOGIWHEELS_FF config option, so it builds for 6.15
+    "KCFLAGS=-DCONFIG_LOGIWHEELS_FF"
   ];
 
   meta = with lib; {

